libs/brig-types/src/Brig/Types/Intra.hs

```diff
@@ -119,8 +119,11 @@ instance FromJSON UserAccount where
 
 instance ToJSON UserAccount where
   toJSON (UserAccount u s) =
-    let Object o = toJSON u
-     in Object $ M.insert "status" (toJSON s) o
+    case toJSON u of
+      Object o ->
+        Object $ M.insert "status" (toJSON s) o
+      other ->
+        error $ "toJSON UserAccount: not an object: " <> show (encode other)
```

Comment:
Hum, I followed the discussion and thought I agreed, but now I'm actually not sure in which sense this is better, am I missing something?  I suppose it's more explicit when reading the code?  Clearer semantics?  That might be enough to make it a good change.

Does the error message at least still contain the source code location?

Reply:
In some cases I managed to avoid the incomplete pattern match, in other (like this one), it wasn't as straight-forward. But yeah, as you said, it's at least much easier to see that this is partial.

I'm not sure if the error message contains the source code location, but it has a greppable string.